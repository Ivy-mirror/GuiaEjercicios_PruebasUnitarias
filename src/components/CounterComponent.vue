<template>
  <div>
    <h2>Contador</h2>
    <p id="contador">{{ contador }}</p>
    <div class="botones">
      <button @click="disminuir" id="disminuir">-1</button>
      <button @click="aumentar" id="aumentar">+1</button>
      <button @click="aumentarDos" id="aumentarDos">+2</button>
      <button @click="disminuirDos" id="disminuirDos">-1</button>
    </div>
  </div>
</template>
<script setup>
import { ref } from 'vue'

const contador = ref(0)

function aumentar() {
  contador.value++
}

function disminuir() {
  contador.value--
}
function disminuirDos() {
  contador.value -= 2
}
function aumentarDos() {
  contador.value += 2
}
</script>
