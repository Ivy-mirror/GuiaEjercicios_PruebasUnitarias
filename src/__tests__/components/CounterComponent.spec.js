import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import CounterComponent from '@/components/CounterComponent.vue'

describe('CounterComponent', () => {
  const wrapper = mount(CounterComponent)

  it('El HTML se mantiene en las pruebas', () => {
    /* Se verfica si el marcado html se mantiene estable entre pruebas  */
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Tiene un botón para aumentar contador con texto "+1"', () => {
    const btnAumentar = wrapper.find('#aumentar')
    expect(btnAumentar.text()).toBe('+1')
  })
  test('Tiene un botón para disminuir contador con texto "-1"', () => {
    const btnDisminuir = wrapper.find('#disminuir')
    expect(btnDisminuir.text()).toBe('-1')
  })

  it('Aumenta el contador en uno al haccer click en el "+1"', async () => {
    const contadorAntesClick = Number(wrapper.find('#contador').text())

    const btnAumentar = wrapper.find('#aumentar')
    await btnAumentar.trigger('click')

    const contadorDespuesClick = Number(wrapper.find('#contador').text())
    expect(contadorAntesClick).toBeLessThan(contadorDespuesClick)
  })

  it('Disminuye el contador en uno al haccer click en el "-1"', async () => {
    const contadorAntesClick = Number(wrapper.find('#contador').text())

    const btnDisminuir = wrapper.find('#disminuir')
    await btnDisminuir.trigger('click')

    const contadorDespuesClick = Number(wrapper.find('#contador').text())
    expect(contadorDespuesClick).toBeLessThan(contadorAntesClick)
  })

  it('Disminuye el contador en 2 al hacer click en -2', async () => {
    const contadorAntesClick = Number(wrapper.find('#contador').text())
    const btnDisminuirDos = wrapper.find('#disminuirDos')
    await btnDisminuirDos.trigger('click')
    const contadorDespuesClick = Number(wrapper.find('#contador').text())

    expect(contadorDespuesClick).toEqual(contadorAntesClick - 2)
  })
  it('Aumenta el contador en 2 al hacer click en +2', async () => {
    const contadorAntesClick = Number(wrapper.find('#contador').text())
    const btnAumentarDos = wrapper.find('#aumentarDos')
    await btnAumentarDos.trigger('click')
    const contadorDespuesClick = Number(wrapper.find('#contador').text())

    expect(contadorDespuesClick).toEqual(contadorAntesClick + 2)
  })
})
