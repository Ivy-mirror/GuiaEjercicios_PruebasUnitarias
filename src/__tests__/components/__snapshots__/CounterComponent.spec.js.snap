// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CounterComponent > El HTML se mantiene en las pruebas 1`] = `
"<div>
  <h2>Contador</h2>
  <p id="contador">0</p>
  <div class="botones"><button id="disminuir">-1</button><button id="aumentar">+1</button><button id="aumentarDos">+2</button><button id="disminuirDos">-1</button></div>
</div>"
`;
